src/shim/crypto.rs

```diff
@@ -132,15 +135,49 @@ impl Signature {
         self.sig_type
     }
 
+    /// Authenticates the message signature using protocol-specific validation:
+    /// - Delegated: Uses the Ethereum message with RLP encoding for signature verification, Verifies message roundtrip integrity
+    /// - BLS/SECP: Standard signature verification
+    pub fn authenticate_msg(
+        &self,
+        eth_chain_id: EthChainId,
+        msg: &SignedMessage,
+        addr: &crate::shim::address::Address,
+    ) -> anyhow::Result<()> {
+        let mut sig = self.clone();
+        let digest: Vec<u8>;
+        match self.sig_type {
```

Comment:
nit: you can also use the following form:
```
let digest = match self.sig_type {
};
```
which is a bit more idiomatic.

Reply:
Done.